low-level/files/opt/seagate/sspl/prerequisites/sspl_rabbitmq_reinit

```diff
@@ -0,0 +1,334 @@
+              f'"^{queue_name}$" \'{{"ha-mode":"all"}}\''
+    print(command)
+    _send_command(command)
+
+def _get_connection_config(section, exchange_key=EXCHANGE_NAME_KEY,
+                           queue_key=QUEUE_NAME_KEY, routing_key=ROUTING_KEY):
+    vhost = get_sspl_config_value(key=VIRT_HOST_KEY, sect=section, default='SSPL')
+    exchange_name = get_sspl_config_value(key=exchange_key, sect=section)
+    queue_name = get_sspl_config_value(key=queue_key, sect=section)
+    routing_key = get_sspl_config_value(key=routing_key, sect=section)
+    username = get_sspl_config_value(key=RPMQ_USERNAME_KEY, sect=section)
+    password = get_sspl_config_value(key=RPMQ_PASSWORD_KEY, sect=section)
+    return vhost, username, password, exchange_name, queue_name, routing_key
+
+def create_queue(vhost, username, password, exchange_name, queue_name, routing_key):
+    creds = pika.PlainCredentials(username, password)
+    connection = pika.BlockingConnection(
+                        pika.ConnectionParameters(
+                            host='localhost',
+                            credentials=creds,
+                            virtual_host=vhost,))
+    channel = connection.channel()
+    try:
+        channel.queue_declare(queue=queue_name, durable=True)
+    except Exception as e:
+        print('Exception in queue declaration: {}'.format(e))
+    try:
+        channel.exchange_declare(
+            exchange=exchange_name,
+            exchange_type='topic',
+            durable=True,
+        )
+    except Exception as e:
+        print('Exception is exchange declaration: {}'.format(e))
+    channel.queue_bind(
+        queue=queue_name,
+        exchange=exchange_name,
+        routing_key=routing_key
+    )
+    connection.close()
+
+def _create_vhost_if_necessary(virtual_host, product):
+    """ Creates the specified vhost (if necessary).
+
+    No action will occur if the vhost already exists.
+
+    @type virtual_host:           string
+    @param virtual_host:          The vhost to create.
+    @type product:                string
+    @param product:               The product for which initialization is
+                                  being done.
+    """
+    if product.lower() in [x.lower() for x in enabled_products]:
+        cmd = [RABBITMQCTL, 'list_vhosts']
+        vhosts = subprocess.check_output(cmd).decode("utf-8").split('\n')
+        print(f'vhosts: {vhosts}')
+        for vhost in vhosts[1:-1]:
+            if vhost == virtual_host:
+                return
+        try:
+            cmd = f'{RABBITMQCTL} add_vhost {virtual_host}'
+            subprocess.check_output(cmd, shell=True, stderr=subprocess.PIPE)
+        except subprocess.CalledProcessError as e:
+            err_msg = e.stderr.decode('utf-8')
+            print(f'Sub process failed: error: {err_msg}')
+            if 'vhost_already_exists' in err_msg:
+                print(f'Not creating new virtual host in RabbitMQ')
```

Comment:
Not creating new RabbitMQ virtual host in as vhosts exists already

Reply:
This will be corrected.